components/compiler/scanner.cpp

```diff
@@ -366,6 +368,13 @@ namespace Compiler
                 {
                     if (mIgnoreNewline)
                         mErrorHandler.warning ("string contains newline character, make sure that it is intended", mLoc);
+                    else if (name.size() == 1 || name.size() == 2 && name[1] == '\r')
```

Comment:
This combination of || and && is ambiguous.

Reply:
Why? && has higher precedence than ||

Reply:
It's still an undefined behavior. And it makes GCC emit a warning.

Reply:
I'm very happy to hear it's not undefined behaviour. I know c++ is a bit loose sometimes, but that would've been scary.

Anyway, brackets added.